main.py

```diff
@@ -30,4 +79,4 @@
     if not golf_game.use_gui:
         golf_game.play_all()
         result = golf_game.get_state()
-        print(result)
+        print(json.dumps(result, cls=NumpyEncoder))
```

Comment:
Do note that this changes main.py to emit the JSON and golf_game.py to add `landing_history` and `ending_history` to the output. This was necessary in order to render the shot history after-the-fact.

If it isn't okay to merge this, I'm open to adding flags/creating a copy of these scripts.